### PR TITLE
Add support for DateTime valuespecification editor

### LINKED ISSUE #842: Bug: DateTime value specification editor produces StrictDate type causing class cast exception
### Similar issues

- [X] I have searched and found no existing similar issues

### How are you using Studio?

Legend Studio

### Current and expected behavior

A property of type `DateTime` should produce a text input for users to type out the dateTime. This is both for filter and parameter instance editor
Currently we show a `StrictDate` value editor for properties of `DateTime`. This causes a class exception when executing. 

### Steps to reproduce

_No response_

### Model data

_No response_

### Environment

```text
Chrome
```

### Possible solution and workaround

_No response_

### Contribution

- [ ] I would like to work on the fix for this issue

### PR DESCRIPTION
**Pull request by gayathrir11**
## Summary

Closes #842 

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx [hunk 1]
```diff
@@ -250,6 +250,31 @@ export const DatePrimitiveInstanceValueEditor = observer(
...
+    const value = valueSpecification.values[0] as string;
+    const changeValue: React.ChangeEventHandler<HTMLInputElement> = (event) => {
+      valueSpecification.changeValue(event.target.value, 0);
+    };
+
+    return (
+      <div className={clsx('query-builder-value-spec-editor', className)}>
+        <input
+          className="panel__content__form__section__input query-builder-value-spec-editor__input"
+          type="datetime-local"
```
Comment (akphi) [inline packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx:268]: @MauricioUyaguari what should it be here? `local` or `UTC`?

File: .changeset/slimy-spiders-hope.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@finos/legend-query": patch
+---
+
+Add support for `DateTime` valueSpecification editor
```
Comment (akphi) [inline .changeset/slimy-spiders-hope.md:5]:
```suggestion
Add support for `DateTime` value specification editor ([#842](https://github.com/finos/legend-studio/issues/842))
```

**Review by akphi — CHANGES_REQUESTED** [review]

**Comment by akphi** [comment]
@gayathrir11 Let's add a quick video for this. Thanks!

**Comment by gayathrir11** [comment]
> @gayathrir11 Let's add a quick video for

> @gayathrir11 Let's add a quick video for this. Thanks!

Screenshot 2022-03-03 172911
 It looks like this. I don't like it.  Thinking of a better design

**Comment by akphi** [comment]
> > @gayathrir11 Let's add a quick video for this. Thanks!
> 
> Screenshot 2022-03-03 172911
> 
> It looks like this. I don't like it. Looking for a better design

It's fine 😛 that's the default from browser. We don't need to ask for more now.

File: packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx (hunk 1 above)
Comment (akphi) [reply packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx:268]: I think we should do `UTC`

**Comment by akphi** [comment]
@gayathrir11 let's add video to the PR description about setting this and going in/out of text mode (kinda a roundtrip)

**Comment by gayathrir11** [comment]
> @gayathrir11 let's add video to the PR description about setting this and going in/out of text mode (kinda a roundtrip)

Cool i will add video but the recording wouldn't show the editor actually and it's in the execute panel so i don't think there will be any changes to the query

**Comment by akphi** [comment]
> > @gayathrir11 let's add video to the PR description about setting this and going in/out of text mode (kinda a roundtrip)
> 
> Cool i will add video but the recording wouldn't show the editor actually and it's in the execute panel so i don't think there will be any changes to the query so nothing will be actually visisble in text mode

Why though? We could show a video when you use this in filter?

File: packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx (hunk 1 above)
Comment (akphi) [reply packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx:268]: Could we try this https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local#setting_timezones
so we set the timezone to UTC?

**Review by akphi — APPROVED** [review]
I'll test out this change myself, if it does not suffice, I can roll back.

File: packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx (hunk 1 above)
Comment (akphi) [reply packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx:268]: Turns out `datetime-local` already assume `UTC` 😆 so we're good I guess

**Comment by akphi** [comment]
Turns out the `datetime-local` actually worked with `UTC` by default - see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local#setting_timezones

We're good folks!

**Pull request merged**